[DOC-2772] [DOC-2875] - Small updates to cache intelligence, hosted infra (#1173)
harness/developer-hub: docs/continuous-integration/use-ci/caching-ci-data/cache-intelligence.md

Currently, Cache Intelligence is available only when using [Harness Cloud](/docs/continuous-integration/ci-quickstarts/hosted-builds-on-virtual-machines-quickstart), the Harness-hosted build environment.

> Currently, Cache Intelligence supports Bazel, Maven, Gradle, Yarn, Go, and Node build tools, if the dependencies are stored in the default location for the tool used.

Reviewer: ```suggestion
Currently, Cache Intelligence supports Bazel, Maven, Gradle, Yarn, Go, and Node build tools. The dependencies must be stored in the default location for the tool used.
```
